go/ql/lib/semmle/go/security/IncorrectIntegerConversionLib.qll

```diff
@@ -229,31 +356,26 @@ private module ConversionWithoutBoundsCheckConfig implements DataFlow::StateConf
     // can sanitize the result of the conversion to prevent flow on to further sinks
     // without needing to use `isSanitizerOut`, which doesn't work with flow states
     // (and therefore the legacy `TaintTracking::Configuration` class).
-    exists(boolean sinkIsSigned, int sinkBitSize |
-      state = TFlowstate(sinkIsSigned, _, sinkBitSize)
-    |
-      isSinkWithBitSize(sink.getASuccessor(), sinkIsSigned, sinkBitSize)
-    )
+    isSink2(sink.getASuccessor(), state)
   }
 
   predicate isBarrier(DataFlow::Node node, FlowState state) {
-    exists(boolean sinkIsSigned, int sourceBitSize, int sinkBitSize |
-      state = TFlowstate(sinkIsSigned, sourceBitSize, sinkBitSize)
-    |
-      // To catch flows that only happen on 32-bit architectures we
-      // consider an architecture-dependent sink bit size to be 32.
-      exists(UpperBoundCheckGuard g, int bitSize |
-        if sinkBitSize != 0 then bitSize = sinkBitSize else bitSize = 32
-      |
-        node = DataFlow::BarrierGuard<upperBoundCheckGuard/3>::getABarrierNodeForGuard(g) and
-        g.isBoundFor(bitSize, sinkIsSigned)
-      )
-      or
-      exists(int bitSize |
-        isIncorrectIntegerConversion(sourceBitSize, bitSize) and
-        isSinkWithBitSize(node, sinkIsSigned, bitSize)
-      )
+    // Safely guarded by a barrier guard.
+    exists(BarrierFlowStateTransformer bfst | node = bfst and bfst.barrierFor(state) |
+      not exists(bfst.transform(state)) or bfst.transform(state) != state
```

Comment:
Presumably it's an implicit rule of the transformer that it shouldn't ever include an identity transform -- for example, we can't / shouldn't write 32 -> [32, 64]